pypeit/spectrographs/ldt_deveny.py

```diff
@@ -0,0 +1,357 @@
+        self.meta['decker'] = dict(card=None, compound=True)
+        self.meta['binning'] = dict(card=None, compound=True)
+        self.meta['mjd'] = dict(card=None, compound=True)
+        self.meta['airmass'] = dict(ext=0, card='AIRMASS')
+        self.meta['exptime'] = dict(ext=0, card='EXPTIME')
+        
+        # Extras for config and frametyping
+        self.meta['idname'] = dict(ext=0, card='IMAGETYP')
+        self.meta['dispangle'] = dict(ext=0, card='GRANGLE', rtol=1e-3)
+        self.meta['filter1'] = dict(card=None, compound=True)
+        self.meta['slitwid'] = dict(ext=0, card='SLITASEC')
+        self.meta['lampstat01'] = dict(card=None, compound=True)
+        
+
+    def compound_meta(self, headarr, meta_key):
+        """
+        Methods to generate metadata requiring interpretation of the header
+        data, instead of simply reading the value of a header card.
+
+        Args:
+            headarr (:obj:`list`):
+                List of `astropy.io.fits.Header`_ objects.
+            meta_key (:obj:`str`):
+                Metadata keyword to construct.
+
+        Returns:
+            object: Metadata value read from the header(s).
+        """
+        if meta_key == 'binning':
+            """
+            Binning in lois headers is space-separated rather than comma-separated.
+            """
+            binspec, binspatial = headarr[0]['CCDSUM'].split()
+            binning = parse.binning2string(binspec, binspatial)
+            return binning
+        
+        elif meta_key == 'mjd':
+            """
+            Use astropy to convert 'DATE-OBS' into a mjd.
+            """
+            ttime = Time(headarr[0]['DATE-OBS'], format='isot')
+            return ttime.mjd
+        
+        elif meta_key == 'lampstat01':
+            """
+            The spectral comparison lamps turned on are listed in `LAMPCAL`, but
+            if no lamps are on, then this string is blank.  Return either the
+            populated `LAMPCAL` string, or 'off' to ensure a positive entry for
+            `lampstat01`.
+            """
+            lampcal = headarr[0]['LAMPCAL'].strip()
+            if lampcal == '':
+                return 'off'
+            else:
+                return lampcal
+        
+        elif meta_key == 'dispname':
+            """
+            Convert older FITS keyword GRATING (gpmm/blaze) into the newer
+            Grating ID names (DVx) for easier identification of disperser.
+            """
+            gratings = {"DV1": "150/5000", "DV2": "300/4000", "DV3": "300/6750",
+                        "DV4": "400/8500", "DV5": "500/5500", "DV6": "600/4900",
+                        "DV7": "600/6750", "DV8": "831/8000", "DV9": "1200/5000",
+                        "DV10": "2160/5000", "DVxx": "UNKNOWN"}
+            ids = list(gratings.keys())
+            kwds = list(gratings.values())
+            idx = kwds.index(headarr[0]['GRATING'])
+            return f"{ids[idx]} ({kwds[idx]})"
```

Comment:
No need to change anything, but this feels round about.  You define the dictionary as key = new name and value = old name, but you want to select based on the old name.  Why not define the dictionary in opposite way so that you return something like `f"{gratings[headarr[0]['GRATING']]} {headarr[0]['GRATING']}"` ?  Regardless, you might want to put in a check that the header keyword is in the list of possibilities, and raise an exception if not.

Reply:
Check added, although the GRATING FITS keyword is limited to a hardwired list of choices, with `UNKNOWN` being inserted if one of the options is not selected.  The dictionary has been flipped, as suggested.